spring-boot/proxy/src/main/java/eu/coatrack/proxy/metrics/MetricsCounterService.java

```diff
@@ -21,131 +21,91 @@
-    public void increment(HttpServletRequest request, String apiKey, MetricType metricType, Integer httpResponseCode) {
+    private void logBeginningOfIncrementation(MetricsHolder mh) {
         log.debug(String.format("incrementing metric '%s' for URI '%s' and api key %s",
-                metricType,
-                request.getRequestURI(),
-                apiKey
+                mh.getMetricType(),
+                mh.getRequestURI(),
+                mh.getApiKeyValue()
         ));
-
-        String requestMethod = request.getMethod();
-
-        String serviceApiName = null;
-        String path = null;
-
-        Matcher matcher = PATTERN_TO_SPLIT_SERVLET_PATH.matcher(request.getServletPath());
-
-        if (matcher.find()) {
-            // first element of the servlet path is the service api's name/id
-            serviceApiName = matcher.group(MATCHER_GROUP_INDEX_OF_SERVICE_API_ID);
-            // rest of the servlet path is the actual path that is called on the proxied service
-            path = matcher.group(MATCHER_GROUP_INDEX_OF_PATH);
-            log.debug("matched servlet path '{}' with service uri identifier '{}' and path '{}'",
-                    matcher.group(0), serviceApiName, path);
-            if (path == null) {
-                // if api consumer did not specifc path, set as "/"
-                path = "/";
-            } else if (path.endsWith("/") && !path.equals("/")) {
-                // remove trailing "/" as this is the same path from metrics point of view
-                path = path.substring(0, path.length()-1);
-            }
-        } else {
-            log.warn("matcher {} did not match servlet path {}", matcher, request.getServletPath());
-        }
-
-        LocalDate today = LocalDate.now();
-
-        counterService.increment(new StringBuilder()
-                .append(PREFIX) // [0]
-                .append(SEPARATOR)
-                .append(serviceApiName) // [1]
-                .append(SEPARATOR)
-                .append(requestMethod) // [2]
-                .append(SEPARATOR)
-                .append(apiKey) // [3]
-                .append(SEPARATOR)
-                .append(metricType) // [4]
-                .append(SEPARATOR)
-                .append(httpResponseCode) // [5]
-                .append(SEPARATOR)
-                .append(today) // [6]
-                .append(SEPARATOR)
-                .append(path) // [7]
-                .toString());
     }
 
-    public Metric filterAndTransformMetric(org.springframework.boot.actuate.metrics.Metric inputMetric) {
-
-        String[] elements = inputMetric.getName().split(SEPARATOR);
-        Assert.noNullElements(elements);
-
-        // check if this is a custom metric
-        if (elements[0].equals("counter." + PREFIX)) {
-            log.debug("Metrics for transmission: " + inputMetric.getName());
+    private int incrementAndReturnCurrentCounter(MetricsHolder mh) {
+        String counterId = createCounterIdFromMetric(mh);
```

Comment:
It would be helpful for debugging purposes to log the generated counter ID, this makes the log also comparable to the old version.
e.g. `log.debug("Incrementing counter with ID {}", counterId)`

Reply:
Done as suggested.